File: Caesar in Python/caesar.py
from cs50 import get_string
from sys import argv


if len(argv) != 2:
    exit("Format: python caesar.py n")

key=int(argv[1])

plain=get_string("plaintext: ")

print("ciphertext:", end="")
for x in plain:
    if x.islower() and (ord(x) + (key % 26) > 122):
        x=chr(ord(x) - (26 - (key % 26)))
        print(f"{x}", end="")
    elif x.islower() and (ord(x) + (key % 26) < 122):
        x=chr(ord(x) + (key % 26))
        print (f"{x}", end="")
    elif  x.isupper() and (ord(x) + (key % 26) > 90):
        x=chr(ord(x) - (26 - (key % 26)))
        print(f"{x}", end="")
    elif  x.isupper() and (ord(x) + (key % 26) < 90):
        x=chr(ord(x) + (key % 26))
        print(f"{x}", end="")
    else:
        print(f"{x}", end="")
print()

File: Caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>


int main(int argc, string argv[])

{
  if (argc != 2)
  {
   printf("Please enter one command line argument!\n");
   return 1;
  }
  else
  {
   string text= get_string("plaintext: ");
   printf("ciphertext: ");
   for (int i=0,n=strlen(text); i < n; i++)
    {
     if (islower(text[i]))
      {
       //Checks if the key makes text[i] become a non-letter(lowercase)
       if (atoi(argv[1]) <= 25 && (int) text[i] + atoi(argv[1]) > 122)
        {
        //If yes, shift text[i] by 26 minus key, to go back to beginning of alphabet
        text[i]=(int) text[i] - (26 -(atoi(argv[1])));
        printf("%c", text[i]);
        }
          //If argument is greater than 26,  modulo (%) then do as above
       else if (atoi(argv[1]) > 26 && ((int) text[i] + (atoi(argv[1]) % 26))>122)
        {
         text[i]=(int) text[i] - (26-(atoi(argv[1]) % 26));
         printf("%c", text[i]);
        }
       else if (atoi(argv[1]) > 26 && ((int) text[i] + (atoi(argv[1]) % 26)) < 122)
        {
         text[i]=(int) text[i] + (26-(atoi(argv[1]) % 26));
         printf("%c", text[i]);
        }
       else
        {
         text[i]=(int) text[i] +atoi(argv[1]);
         printf("%c", text[i]);
        }
      }
     else if (isupper(text[i]))
      {
       //Checks if the key makes text[i] become a non-letter(Uppercase)
       if (atoi(argv[1]) <= 25 && (int) text[i] + atoi(argv[1]) > 90)
        {
         //If yes, shift text[i] by 26 minus key, to go back to beginning of alphabet
         text[i]=(int) text[i] - (26 -(atoi(argv[1])));
         printf("%c", text[i]);
        }
         //If argument is greater than 26,  modulo (%) then do as above
        else if (atoi(argv[1]) > 26 && ((int) text[i] + (atoi(argv[1]) % 26)) > 90)
         {
          text[i]=(int) text[i] - (26-(atoi(argv[1]) % 26));
          printf("%c", text[i]);
         }
        else if (atoi(argv[1]) > 26 && ((int) text[i] + (atoi(argv[1]) % 26)) < 90)
         {
         text[i]=(int) text[i] + (26-(atoi(argv[1]) % 26));
          printf("%c", text[i]);
         }
        else
         {
          text[i]=(int) text[i] +atoi(argv[1]);
          printf("%c", text[i]);
         }
       }
      else
       {
        printf("%c", text[i]);
       }
    }
    printf("\n");
    return 0;
   }
 }
